src/apps/backend/modules/communication/sms-service.ts

```diff
@@ -1,8 +1,20 @@
+import { VerificationInstance } from 'twilio/lib/rest/verify/v2/service/verification';
+
+import CommunicationUtil from './communication-util';
 import TwilioService from './internals/twilio-service';
-import { SendSMSParams } from './types';
+import { SendSMSParams, VerifyOtpResponse } from './types';
 
 export default class SMSService {
   public static async sendSMS(params: SendSMSParams): Promise<void> {
     return TwilioService.sendSMS(params);
   }
+
+  public static async sendOTP(contactNumber: string): Promise<VerificationInstance> {
+    CommunicationUtil.validatePhoneNumber(contactNumber);
+    return TwilioService.sendOTP(contactNumber);
+  }
+
+  public static async verifyOTP(contactNumber: string, otp: string): Promise<VerifyOtpResponse> {
+    return TwilioService.verifyOTP(contactNumber, otp);
+  }
```

Comment:
Didn't get this one. Are we not going to utilize an external service?

Is this the flow we are thinking when it is OTP based signup/login

POST `/otps` (create OTP Record) (Generating 6-digit OTP randomly for now)
PATCH `/otps` (verify OTP from OTP record)
POST `/accounts` (Signing up by contact number)
POST `/access-tokens` (Signing in)

Reply:
Let's do this:

**Email based auth** (existing flow):

POST `/accounts` (to create account)
body params: email, password

POST `/access-tokens` (to login)
body params: email, password

**OTP based auth** (new flow):

POST `/accounts` (to create account)
body params: phone number

Note:
- Unlike sign up, this will not throw if account exists with the phone number.
- System sends OTP (system stores in `otps` collection and store code (4 digit) and phoneNumber, and use Twilio SMS product to sms the OTP)

POST `/access-tokens` (to login)
body params: phone number, OTP

Note:
- System verifies again the `otps` collection to match the code. If Ok, sends access token